rdflib/term.py

```diff
@@ -606,7 +608,12 @@ def __new__(
             # passed a string
             # try parsing lexical form of datatyped literal
             value = _castLexicalToPython(lexical_or_value, datatype)
-
+            if datatype:
+                dt_uri: URIRef = URIRef(datatype)
+                if dt_uri in _check_well_formed_types:
+                    checker = _check_well_formed_types[dt_uri]
+                    well_formed = checker(lexical_or_value, value)
+                    ill_formed = ill_formed or (not well_formed)
```

Comment:
are there cases where this value will be different from:

```suggestion
                    ill_formed = not well_formed
```

Reply:
No, for now there are no instances where it will be different. But this is a coding pattern I've picked up from other projects, where other code may be added in the future that will set the boolean before this point. Probably not needed for now though.